### PR TITLE
Makefile

### PR DESCRIPTION
**Pull request by benja-wu**
* Add md/html/json file, and Makefile
* Docs are generated by photoc-gen-doc, chapters are sorted alphabetically by this tool

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+HTML_FILENAME := meshmodel.html
+GO_FILENAME := meshmodel.pb.go
+JSON_FILENAME := meshmodel.json
+
+PROTO := $(shell find ${MKFILE_DIR}${RELEASE} -type f -name "*.proto") 
+
+go: ${PROTO} 
+	@echo "generate golang file from proto: ${PROTO} "
+		@echo ""
+	protoc -I ${MKFILE_DIR}${RELEASE} --go_out=${MKFILE_DIR}${RELEASE}/ ${PROTO}  && \
```
Comment (zhao-kun) [inline Makefile]: Where can I find the `protoc`?  You should not assume users have installed it!

Comment (benja-wu) [reply Makefile]: I though this dependences should be mentioned in README. Let me prepare it at the next PR?

Comment (zhao-kun) [reply Makefile]: Can you refer to Makefile of others? I think that Makefile is dedicated to managing dependencies

Comment (benja-wu) [reply Makefile]: Actually, I referred to EG's Makefile as an example, in EG's Makefile we had made an assumption: the user is working in the Golang environment. So in the mesh-api project, I thought we could just tell users how to install `protoc` in README.
Or we can use docker container cmd for using `protoc` in Makefile, but is that mean we should check and install docker for users? 
And I had searched the [Istio's API repo](https://github.com/istio/api/search?p=2&q=protoc), still can't find the `protoc` installing step...
I am confused here.

Comment (zhao-kun) [reply Makefile]: Did you seriously or carefully check [Istio's API repo](https://github.com/istio/api), even the Istio has its own solution, it still need to handle dependencies.

Comment (benja-wu) [reply Makefile]: Already updated to check `protoc`, `protoc-gen-doc` and `protoc-gen-go` dependence.

**Review by zhao-kun — APPROVED** [review]

**Pull request merged**